src/autosize.js

```diff
@@ -127,21 +127,29 @@ function assign(ta) {
 	function update() {
 		resize();
 
-		const styleHeight = Math.round(parseFloat(ta.style.height));
+    const scrollHeight = ta.scrollHeight;
+    const styleHeight = Math.round(parseFloat(ta.style.height));
 		const computed = window.getComputedStyle(ta, null);
 
 		// Using offsetHeight as a replacement for computed.height in IE, because IE does not account use of border-box
 		var actualHeight = computed.boxSizing === 'content-box' ? Math.round(parseFloat(computed.height)) : ta.offsetHeight;
 
-		// The actual height not matching the style height (set via the resize method) indicates that 
+		// The actual height not matching the style height (set via the resize method) indicates that
 		// the max-height has been exceeded, in which case the overflow should be allowed.
 		if (actualHeight < styleHeight) {
 			if (computed.overflowY === 'hidden') {
 				changeOverflow('scroll');
 				resize();
 				actualHeight = computed.boxSizing === 'content-box' ? Math.round(parseFloat(window.getComputedStyle(ta, null).height)) : ta.offsetHeight;
 			}
-		} else {
+		}
+    // If the scrollHeight is higher than the styleHeight at this point, it means that the textearea is taller than its parent and a scrollbar appeared
```

Comment:
There is a potential miscalculation of the textarea's height in the `resize()` method. 


Let's assume that the parent container is 300px wide and 400px long. Let's assume we paste a large text making the textarea 1200px height.

```
// because of this line, the textarea goes back to its initial size
// this is usually the size when it was empty
// The parent container will therefore remove its scrollbar and the textarea will be the full width, so 300px.
ta.style.height = '';

// the scrollHeight value here is calculated based on the textarea  being 300px.
ta.style.height = (ta.scrollHeight+heightOffset)+'px';

// at this point, the content is added to the DOM, the textarea has an height of 1200px (as calculated by the scrollHeight). Because the parent's height is 400px, a scrollbar gets added. The textarea is now 300px - 15/20px (the width of the scrollbar).

This will cause a shift in layout as some words will now break into a new line. This means that the scrollHeight is now higher as the width of the textarea is smaller.
```

We cannot fix it in the `resize()` method so I override the value in the `update` method.